### PR TITLE
fix:  Add support for SET OF to Copy As ECL

### LINKED ISSUE #235: Incorrect "Copy All As ECL" result
Given a result with layout {STRING parent_name, SET OF STRING children}, the "Copy All As ECL" feature produces the following incorrect (snipped) result:

r := RECORD
    VARSTRING parent_name;
    VARSTRING children;
END;

d := DATASET([
    {'Matthew', },
    {'Kenneth', },
    {'Walter', },
    ...
    {'Margaret', },
    {'Alice', },
    {'Eric', },
    {'Andrew', },
    {'Donald', },
    {'Peter', },
    {'Frank', }
], r);

Enclosed file is actually an ECL BWR that generates random results in that layout.

[sample.txt](https://github.com/hpcc-systems/vscode-ecl/files/6084012/sample.txt)

### PR DESCRIPTION
**Pull request by GordonSmith**
Fixes #235

Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

### CODE REVIEW

File: src/eclwatch/WUResult.tsx [hunk 1]
```diff
@@ -12,22 +12,23 @@ import { Store } from "./WUResultStore";
...
+            return prefix + "REAL";
         case "xs:string":
-            return "VARSTRING";
+            return prefix + "VARSTRING";
         case "xs:hexBinary":
-            return "DATA";
+            return prefix + "DATA";
         default:
-            return type.toUpperCase();
+            return prefix + type.toUpperCase();
```
Comment (jbrundage) [inline src/eclwatch/WUResult.tsx:31]: There's no chance the type could already have a prefix of "SET OF "?

File: src/eclwatch/WUResult.tsx [hunk 2]
```diff
@@ -73,7 +79,7 @@ function copyColumnTPL(col: number, dedup: boolean, fields: XSDXMLNode[], row: o
         set = row.map(r => valueTPL(r[name]));
     }
     return `
-SET OF ${typeTPL(fields[col].type)} ${name} := [${set.join(",")}];
+SET OF ${typeTPL(fields[col].type, fields[col].isSet)} ${name} := [${set.join(",")}];
```
Comment (jbrundage) [inline src/eclwatch/WUResult.tsx:82]: Could this result in "SET OF SET OF ___"?

**Review by jbrundage — APPROVED** [review]
Looks good if the "SET OF SET OF" comment isn't a concern.

File: src/eclwatch/WUResult.tsx (hunk 2 above)
Comment (GordonSmith) [reply src/eclwatch/WUResult.tsx:82]: Nope - the same way we aren't getting a "STRING STRING"...

Comment (GordonSmith) [reply src/eclwatch/WUResult.tsx]: @jbrundage - I "fixed" that, please re-review.

**Review by dehilsterlexis — APPROVED** [review]
@GordonSmith looks good

**Review by jbrundage — APPROVED** [review]

**Pull request merged**